sonic_installer/main.py

```diff
@@ -208,6 +208,50 @@ def print_deprecation_warning(deprecated_cmd_or_subcmd, new_cmd_or_subcmd):
                 fg="red", err=True)
     click.secho("Please use '{}' instead".format(new_cmd_or_subcmd), fg="red", err=True)
 
+def update_sonic_environment(click, binary_image_version):
+    """Prepare sonic environment variable using incoming image template file. If incoming image template does not exist
+       use current image template file.
+    """
+    def mount_next_image_fs(squashfs_path, mount_point):
+        run_command_or_raise(["mkdir", "-p", mount_point])
+        run_command_or_raise(["mount", "-t", "squashfs", squashfs_path, mount_point])
+
+    def umount_next_image_fs(mount_point):
+        run_command_or_raise(["umount", "-rf", mount_point])
+        run_command_or_raise(["rm", "-rf", mount_point])
+
+    SONIC_ENV_TEMPLATE_FILE = os.path.join("usr", "share", "sonic", "templates", "sonic-environment.j2")
+    SONIC_VERSION_YML_FILE = os.path.join("etc", "sonic", "sonic_version.yml")
+
+    sonic_version = re.sub("SONiC-OS-", '', binary_image_version)
+    new_image_dir = os.path.join('/', "host", "image-{0}".format(sonic_version))
+    new_image_squashfs_path = os.path.join(new_image_dir, "fs.squashfs")
+    new_image_mount = os.path.join('/', "tmp", "image-{0}-fs".format(sonic_version))
+
+    try:
+        mount_next_image_fs(new_image_squashfs_path, new_image_mount)
+
+        next_sonic_env_template_file = os.path.join(new_image_mount, SONIC_ENV_TEMPLATE_FILE)
+        next_sonic_version_yml_file = os.path.join(new_image_mount, SONIC_VERSION_YML_FILE)
+
+        sonic_env = run_command_or_raise([
+                "sonic-cfggen",
+                "-d",
+                "-y",
+                next_sonic_version_yml_file,
+                "-t",
+                next_sonic_env_template_file,
+        ])
+        env_dir = os.path.join(new_image_dir, "sonic-config")
+        os.mkdir(env_dir, 0o755)
+        env_file = os.path.join(env_dir, "sonic-environment")
+        with open(env_file, "w+") as ef:
+            print >>ef, sonic_env
+        os.chmod(env_file, 0o644)
+    except Exception as ex:
```

Comment:
>Exception [](start = 11, length = 9)

It is recommended not to throw and capture very general exception type. It will hide other possible exceptions in this block.

Reply:
I thought your ask was to bail out when this module errs. I can print the exception and bail out.